pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1788,6 +1788,46 @@ func (s *TracerSuite) TestPreexistingConnectionDirection() {
 	assert.Equal(t, network.INCOMING, incoming.Direction)
 }
 
+func (s *TracerSuite) TestPreexistingEmptyIncomingConnectionDirection() {
+	t := s.T()
+	// Start the client and server before we enable the system probe to test that the tracer picks
+	// up the pre-existing connection
+
+	ch := make(chan struct{})
+	server := NewTCPServer(func(c net.Conn) {
+		<-ch
+		c.Close()
+	})
+	require.NoError(t, server.Run())
+	t.Cleanup(server.Shutdown)
+
+	c, err := net.DialTimeout("tcp", server.address, 50*time.Millisecond)
```

Comment:
nit, is `50ms` enough to make a connection given the potential slowness of a busy CI env?

Reply:
It may not be; fixed